src/Export/ProductService.php

```diff
@@ -169,7 +174,8 @@ protected function getChildrenOrSiblings(ProductEntity $product): ?ProductCollec
 
         // Remove the given children, as the child product is considered as the product, which is shown
         // in the storefront. As we also want to get the data from the parent, we also manually add it here.
-        $children = $result->first()->getChildren();
+        $mainProduct = clone $result->first();
+        $children = $mainProduct->getChildren();
```

Comment:
Not using a cloned instance of the product, lead to the children being recursively. So it was possible to do this:
`$result->first()->getChildren()->last()->getChildren()->last()->getChildren()->last()->getChildren()->last()`

Before:
- Requested Variant
    - Sibling 1
    - Sibling 2
    - Main Variant
        - Sibling 1 / Child 1
        - Sibling 2 / Child 2
        - Main variant
            - ....

Now:
- Requested Variant
    - Sibling 1
    - Sibling 2
    - Main Variant
       - Sibling 1 / Child 1
       - Sibling 2 / Child 2